File: Algorithms/Sorting/Insertion_Sort/Insertion_Sort.java
public class Insertion_Sort{

     public static void main(String[] args) {
          
          int[] a = {12, 11, 3, 2, 5, 7};
          sort(a, a.length);
          for (int i = 0; i < a.length; i++)
          {
               System.out.print(a[i] + " ");
          }
     }
     public static void sort(int[] a,int length) {
          
          int n = a.length;

          for (int i = 0; i < n; i++)
          {
               int key = a[i];
               //start of sorted region
               int j = i - 1;
               while (j >= 0 && a[j] > key)
               {
                    //shift
                    a[j + 1] = a[j];
                    j = j - 1;
               }
               a[j + 1] = key;
          }
     }
}

File: Algorithms/Sorting/Insertion_Sort/insertion_sort.cpp
// Ascending order
// Insertion Sort

#include <bits/stdc++.h>

using namespace std;

void insertion_sort(int arr[], int l);
int main()
{
	int n;
    cout<<"Enter number of elements: ";
	cin>>n;
	int arr[n], i, l;
	cout<<"\nEnter the elements of array: ";
	for(i=0; i<n; i++)
	{
		cin>>arr[i];
	}
	cout<<"\nThe given array is : "<<endl;
	for(i=0; i<n; i++)
	{
		cout<<arr[i]<<" ";
	}
	insertion_sort(arr, n);
	cout<<"\n\n\n The sorted array is : "<<endl;
	for(i=0; i<n; i++)
	{
		cout<<arr[i]<<" ";
	}

    return 0;
}

void insertion_sort(int arr[], int l)
{
	int i, j, current;
	for(i=1; i<l; i++)
	{
		current = arr[i];
		j = i;
		
		while(j>0 && current<arr[j-1])
		{
			arr[j] = arr[j-1];
			arr[j-1] = current;
			j--;
		}
	}
	
}

